### PR TITLE
Feature/controllers and views

### PR DESCRIPTION
**Pull request by tariq-9484**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -17,6 +17,12 @@ gem 'webpacker', '~> 4.0'
 gem 'turbolinks', '~> 5'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 gem 'jbuilder', '~> 2.7'
+gem 'pry'
+gem 'jquery-rails'
+gem 'jquery-ui-rails'
```
Comment (hamzashariq7) [inline Gemfile:22]: Remove this and all other changes you made when installing jquery, and install it through yarn instead.

File: Gemfile [hunk 2]
```diff
@@ -17,6 +17,12 @@ gem 'webpacker', '~> 4.0'
 gem 'turbolinks', '~> 5'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 gem 'jbuilder', '~> 2.7'
+gem 'pry'
```
Comment (hamzashariq7) [inline Gemfile:20]: Move this inside `group: development`

File: app/controllers/medicines_controller.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
+class MedicinesController < ApplicationController
+	include ActionController::MimeResponds
+	def index
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb:3]: Add a line here.

File: app/controllers/medicines_controller.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		@medicine = Medicine.all
+	end
+	def new
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb:6]: Add lines between function definitions.

File: app/controllers/medicines_controller.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+		@medicine = Medicine.new
+	end
+	def show
+		@medicine = Medicine.find(params[:id])
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb:10]: Make sure indentation is perfect everywhere in your code; should be two spaces only.

File: app/controllers/medicines_controller.rb [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+	end
+
+	def search
+		#binding.pry
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb:14]: remove this.

File: db/schema.rb [hunk 7]
```diff
@@ -0,0 +1,203 @@
+# This file is auto-generated from the current state of the database. Instead
```
Comment (hamzashariq7) [inline db/schema.rb:1]: Add this file to git ignore.

File: config/routes.rb [hunk 8]
```diff
@@ -1,3 +1,20 @@
 Rails.application.routes.draw do
+
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  resources :medicines do
+  	collection do
+    	get 'search'
+    end
+    member do
+    	post 'decquantity'
```
Comment (hamzashariq7) [inline config/routes.rb:9]: What does this lead to?

File: config/routes.rb [hunk 9]
```diff
@@ -1,3 +1,20 @@
 Rails.application.routes.draw do
+
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  resources :medicines do
+  	collection do
+    	get 'search'
+    end
```
Comment (hamzashariq7) [inline config/routes.rb:7]: Fix indentation.

File: config/routes.rb [hunk 10]
```diff
@@ -1,3 +1,20 @@
...
+    member do
+    	post 'decquantity'
+    end
+  end
+  root 'medicines#index'
```
Comment (hamzashariq7) [inline config/routes.rb:12]: This will not be our global root, so remove this.

File: app/assets/stylesheets/medicines.scss [hunk 11]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Medicines controller here.
```
Comment (hamzashariq7) [inline app/assets/stylesheets/medicines.scss:1]: Remove all these `.scss` files.

File: app/controllers/medicines_controller.rb [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+			redirect_to root_path
+		else		
+			respond_to do |format|
+        		format.js{ render 'purchase_order/search' }
+      	    end
+		end
+	end
+	def create
+		@medicine = Medicine.new(medicine_params)
+		@medicine.hospital=Hospital.first
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb:27]: Add a `todo: ` comment here to do this dynamically once the devise PR has been merged.

File: app/controllers/medicines_controller.rb [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+	    if @medicine.save   
+	      flash[:notice] = 'Medicine successfully added!'   
+	      redirect_to root_path   
+	    else   
+	      flash[:error] = 'Failed to add medicine'   
+	      render :new   
+	    end   
+	end
+	def edit
+		binding.pry
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb:37]: Remove.

File: app/controllers/medicines_controller.rb [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+			respond_to do |format|
+        		format.js{ render 'purchase_order/search' }
+      	    end
+		end
+	end
+	def create
+		@medicine = Medicine.new(medicine_params)
+		@medicine.hospital=Hospital.first  
+	    if @medicine.save   
+	      flash[:notice] = 'Medicine successfully added!'
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb:29]: Move all messages to en.yml

File: app/controllers/purchase_order_controller.rb [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+  def index
+    @purchase_orders = PurchaseOrder.all    
+  end
+
+  def new
+    @purchase_order = PurchaseOrder.new
+  end
+
+  def addmed
+    binding.pry
```
Comment (hamzashariq7) [inline app/controllers/purchase_order_controller.rb:11]: Remove all `binding.prys`.

File: app/models/purchase_order.rb [hunk 16]
```diff
@@ -1,6 +1,13 @@
 class PurchaseOrder < ApplicationRecord
+	sequenceid :hospital , :purchase_orders
+  validates :vendorname, presence: { message: "must be given" }
+  validates :price, presence: true, numericality: {:greater_than => 0}
+  validates :state, presence: true 
   belongs_to :hospital
   belongs_to :admin
   has_many :purchase_details, dependent: :destroy
   has_many :medicines, through: :purchase_details
+  def add_med(medicine)
```
Comment (hamzashariq7) [inline app/models/purchase_order.rb:10]: Try to use full names where possible, `add_medicine` would have been fine.

File: app/controllers/purchase_order_controller.rb [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+    @purchase_order = PurchaseOrder.find(params[:id])
+    @medicine= Medicine.find_by(id: params[:medicine_id])
+    if @purchase_order.add_med(@medicine) 
+        if @medicine.update(quantity: @medicine.quantity-1)
```
Comment (hamzashariq7) [inline app/controllers/purchase_order_controller.rb:15]: Perform this update inside `add_med`, and also check if the medicines quantity is above 0 or not.

File: app/controllers/purchase_order_controller.rb [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+    @purchase_order = PurchaseOrder.find(params[:id])
+  end
+  def edit
+    binding.pry
+    @purchase_order = PurchaseOrder.find(params[:id])
+  end
+  def create
+    @purchase_order = PurchaseOrder.new(purchase_order_params)
+    @purchase_order.hospital=Hospital.first 
+    @purchase_order.admin=User.first
```
Comment (hamzashariq7) [inline app/controllers/purchase_order_controller.rb:39]: Add comments here as well to remember to fix this once the devise PR is merged.

File: app/controllers/purchase_order_controller.rb [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+    binding.pry
+    @purchase_order = PurchaseOrder.find(params[:id])
+  end
+  def create
+    @purchase_order = PurchaseOrder.new(purchase_order_params)
+    @purchase_order.hospital=Hospital.first 
+    @purchase_order.admin=User.first  
+    if @purchase_order.save!  
+        flash[:notice] = 'Order successfully added!'   
+        redirect_to purchase_order_index_path
```
Comment (hamzashariq7) [inline app/controllers/purchase_order_controller.rb:42]:
```suggestion
        redirect_to @purchase_order  
```

File: app/controllers/medicines_controller.rb [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+        		format.js{ render 'purchase_order/search' }
+      	    end
+		end
+	end
+	def create
+		@medicine = Medicine.new(medicine_params)
+		@medicine.hospital=Hospital.first  
+	    if @medicine.save   
+	      flash[:notice] = 'Medicine successfully added!'   
+	      redirect_to root_path
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb:30]:
```suggestion
	      redirect_to @medicine
```

File: app/controllers/medicines_controller.rb [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+	end
+	def edit
+		binding.pry
+		@medicine= Medicine.find(params[:id])
+	end
+	def update
+	  @medicine= Medicine.find(params[:id])   
+	    if @medicine.update_attributes(medicine_params)   
+	      flash[:notice] = 'Med updated!'   
+	      redirect_to root_path
```
Comment (hamzashariq7) [inline app/controllers/medicines_controller.rb:44]: Redirect the user to the updated medicine.

File: app/controllers/purchase_order_controller.rb [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+        render :new  
+    end
+  end
+
+
+  def update
+    @purchase_order= PurchaseOrder.find(params[:id])   
+    if @purchase_order.update_attributes(purchase_order_params)   
+        flash[:notice] = 'Order updated!'   
+        redirect_to purchase_order_index_path
```
Comment (hamzashariq7) [inline app/controllers/purchase_order_controller.rb:54]: Redirect to the purchase order.

File: app/controllers/purchase_order_controller.rb [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+        redirect_to purchase_order_index_path          
+    else   
+        flash[:error] = 'Failed!'   
+        render :edit   
+    end   
+  end
+
+  def destroy
+    @purchase_order = PurchaseOrder.find(params[:id])   
+    if @purchase_order.delete
```
Comment (hamzashariq7) [inline app/controllers/purchase_order_controller.rb:63]:
```suggestion
    if @purchase_order.destroy 
```

File: app/controllers/purchase_order_controller.rb [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+  end
+
+  def destroy
+    @purchase_order = PurchaseOrder.find(params[:id])   
+    if @purchase_order.delete 
+        flash[:notice] = 'Order deleted!'   
+        redirect_to purchase_order_index_path    
+    else   
+        flash[:error] = 'Failed to delete!'   
+        render :destroy
```
Comment (hamzashariq7) [inline app/controllers/purchase_order_controller.rb:68]: What will this render?

File: app/controllers/medicines_controller.rb (hunk 4 above)
Comment (tariq-9484) [reply app/controllers/medicines_controller.rb:6]: already done that in the code

File: app/controllers/medicines_controller.rb (hunk 14 above)
Comment (tariq-9484) [reply app/controllers/medicines_controller.rb:29]: already did that. i guess you somehow got the previous version

**Pull request merged**